### ISSUE TITLE
Add query update summary to yielded data from apoc.periodic.iterate()

### ISSUE THREAD

**Issue by InverseFalcon** [neo4j-contrib/neo4j-apoc-procedures#1815]
## Feature description (Mandatory)
apoc.periodic.iterate() provides only minimal information about the result of the call. Specifically, aside from error info, it provides the total number of rows processed (from the outer driving query) and the total number of batches executed, but it doesn't provide any information of what was updated within those batches.

It would be very useful if update statistics (nodes/rels created, updated, deleted) were obtained during batch execution and summed up so we could YIELD such information.

In regular queries, we get metadata in the summary such as this:

```
"updateStatistics": {
    "_stats": {
      "nodesCreated": 4,
      "nodesDeleted": 0,
      "relationshipsCreated": 0,
      "relationshipsDeleted": 0,
      "propertiesSet": 4,
      "labelsAdded": 4,
      "labelsRemoved": 0,
      "indexesAdded": 0,
      "indexesRemoved": 0,
      "constraintsAdded": 0,
      "constraintsRemoved": 0
    }
}
```

Theoretically we should be able to get these counts from each batch executed, and sum up the counts (while properly omitting on error). This wouldn't be a bad way to YIELD the sums (though probably without the extra nesting), though I can't recall if we can perform schema operations using iterate() or not. If not, then we won't need those last 4.

## Considered alternatives
None at this time.

## How this feature can improve the project?
Batch and total counts currently available from iterate() do not adequately track what was changed in the db by the procedure. This addition would provide much more accurate information, and would provide a very quick way for a user to determine whether any changes were actually made to the db (which could alert them to a flaw in their query quickly), and what those were.

Requesting this for at least APOC 4.1.x.x up

**Comment by voutilad** [comment]
I've got something pretty close to implemented, but found out adding schema updates (indexes, contraints) might be moot as there's no easy way to do schema operations in the producer Cypher statement. Even if you use `apoc.schema.assert()` that proc doesn't return a QueryStatistics map with counts, either!

**Comment by conker84** [comment]
closed via e1939eb
